tools/autograd/gen_variable_type.py

```diff
@@ -185,47 +202,74 @@ def should_trace(declaration):
     return True
 
 
-def record_trace_outputs(declaration):
-    if declaration['name'].endswith('_out'):
+def record_trace_outputs(declaration, try_outplace):
+    if try_outplace and declaration['name'].endswith('_out'):
         output_names = [arg['name'] for arg in declaration['arguments'] if arg.get('output', False)]
     else:
         output_names = [r['name'] for r in declaration['returns']]
     return ['jit::tracer::addOutput(node, {});'.format(n) for n in output_names]
 
 
-def format_trace(declaration):
+def prerecord_trace(declaration, try_outplace):
     local = {}
-    local['trace_name'] = trace_name = uninplace_api_name(declaration['api_name'])
+    if try_outplace:
+        trace_name = uninplace_api_name(declaration['api_name'])
+    else:
+        trace_name = declaration['api_name']
+
+    if trace_name.endswith('_out'):
+        trace_name = trace_name[:-4]
+
+    local['trace_name'] = trace_name
 
-    # *_out functions take the result as a first argument, but since we're
-    # going to de-inplace the call, we need to remove it from the argument list
     trace_inputs = declaration['arguments']
+
     if declaration['name'].endswith('_out'):
+        # *_out functions take the result as a first argument, but since we're
+        # going to de-inplace the call, we need to remove it from the argument list
```

Comment:
So in-place operations in the JIT take the out argument as the last one, unlike ATen?

Reply:
yeah, here's an example schema:
```
      "aten::add(Tensor self, Tensor other, *, Scalar alpha=1, Tensor(a!) out) -> Tensor(a!)",
```